### PR TITLE
Adapt Freds claim generator

### PR DESCRIPTION
**Pull request by palango**
## Description

Adapt Freds claim generator.

- Adds `chain_id` and `token_network_address` to the claims.

Part of https://github.com/raiden-network/raiddit/issues/31

### CODE REVIEW

**Comment by fredo** [comment]
What are the differences in just putting the channel_id into the claim instead of `chain_id`, `token_network_address` and `partner_address` ? 

With the latest approach of @karlb to never close a channel this would be equivivalent to it, wouldnt it?

**Comment by palango** [comment]
> What are the differences in just putting the channel_id into the claim instead of `chain_id`, `token_network_address` and `partner_address` ?

The claim needs to contains the raw data, not the hash. Otherwise the owner wouldn't know for which token network and partner a given claim is valid.

File: tools/raiddit/generate_claims.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    total_amount: TokenAmount
+    signature: Optional[Signature] = None
+
+    def pack(self) -> bytes:
+        return (
+            Web3.toBytes(hexstr=to_hex_address(self.token_network_address))
+            + encode_single("uint256", self.chain_id)
+            + Web3.toBytes(hexstr=to_hex_address(self.owner))
+            + Web3.toBytes(hexstr=to_hex_address(self.partner))
+            + encode_single("uint256", self.total_amount)
```
Comment (karlb) [inline tools/raiddit/generate_claims.py:35]: These encoding functions are not very fast, but that's irrelevant in this case.

File: tools/raiddit/generate_claims.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        )
+
+
+@click.group(name="generate")
+def generate():
+    pass
+
+
+@generate.command()
+@click.option("--address", help="Own address", required=True, type=ADDRESS_TYPE)
```
Comment (karlb) [inline tools/raiddit/generate_claims.py:58]: We could allow this option multiple times to create a file with multiple "useful" claims.

File: tools/raiddit/generate_claims.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        claim.sign(signer)
+        claims.append(claim.serialize())
+
+        reverse_claim = Claim(
+            chain_id=chain_id,
+            token_network_address=token_network_address,
+            owner=hub_address,
+            partner=address,
+            total_amount=DEFAULT_TOKEN_AMOUNT,
+        )
```
Comment (karlb) [inline tools/raiddit/generate_claims.py:99]: We probably don't want to make the hub rich this way, but this is useful for testing.

Comment (fredo) [reply tools/raiddit/generate_claims.py:99]: At least as an answer from reddit, they would run a hub and provide liquidity for it. Anyway, the command is specifically for creating a hub solution. We should definitely implement other commands to generate other topologies in the future.

**Pull request merged**